packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol

```diff
@@ -258,6 +258,9 @@ contract DisputeGameFactory is OwnableUpgradeable, ISemver {
     /// @param _gameType The type of the DisputeGame.
     /// @param _impl The implementation contract for the given `GameType`.
     function setImplementation(GameType _gameType, IDisputeGame _impl) external onlyOwner {
+        if (_gameType.raw() != _impl.gameType().raw()) {
+            revert IncorrectGameType();
+        }
```

Comment:
I wonder why we need the gametype arg at all in the first place? Its possible that things now depend on this interface but it seems unnecessary based on this

Reply:
Yeah we can remove it, but the gametype arg can be used for the double check introduced in this PR.